src/main/java/bimo/Storage.java

```diff
@@ -0,0 +1,129 @@
+package bimo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import bimo.tasks.Deadline;
+import bimo.tasks.Event;
+import bimo.tasks.Task;
+import bimo.tasks.ToDo;
+
+/**
+ * Deals with writing and loading files.
+ */
+public class Storage {
+
+    private File dataFile;
+
+    public Storage(String filePath) {
+        this.dataFile = new File(filePath);
+    }
+    /**
+     * Converts the type of task, status, description and dates into text.
+     *
+     * @param task Task object inside Tasklists.
+     * @return Formatted text in the form type|1|description|yyyy-mm-dd/yyyy-mm-dd.
+     */
+    public String convertTaskToText(Task task) {
+        String text = "";
+        String description = task.getTaskText();
+        if (task instanceof ToDo) {
+            text = "T|" + description;
+        } else if (task instanceof Deadline) {
+            //safe to cast since type checking is done
+            text = "D|" + description + ((Deadline) task).getDateAsText();
+        } else {
+            text = "E|" + description + ((Event) task).getDatesAsText();
+        }
+        return text;
+    }
+
+
+    /**
+     * Converts text to its corresponding Task object.
+     *
+     * @param text line of code inside data/Bimo.txt.
+     * @return Task object.
+     */
+    private Task convertTextToTask(String text) {
+        String[] details = text.split("\\|");
+        String type = details[0];
+        Task task = null;
+
+        boolean status = details[1].equals("0") ? false : true;
+        if (type.equals("T")) {
+            task = new ToDo(details[2]);
```

Comment:
Would writing out: " String description = details[2] " etc etc make the code more readable?